### PR TITLE
411 build issues macos

### LINKED ISSUE #411: Resolve build issues with macOS/clang
Bito currently has build issues when building in clang and in macOS environments.

### PR DESCRIPTION
**Pull request by davidrich27**
## Description

Fix Build Issues with macOS/clang

* Various fixes for apple-clang compiler quirks.
* Also adds the option to disable -Werror at a new make target using "make work".   For bypassing build warnings during development.

Closes #411 

## Tests

N/A

## Checklist:

* [x] Code follows our detailed [contribution guidelines](CONTRIBUTING.md)
* [x] `clang-format` has been run
* [x] TODOs have been eliminated from the code
* [x] Comments are up to date, document intent, and there are no commented-out code blocks

### CODE REVIEW

**Comment by davidrich27** [comment]
This PR is being resurrected because after merging main, it failed the CI build due to some warnings.  So it was reverted.  But issues are now resolved, passes builds on Quokka, macOS, and Github CI.

**Comment by ognian-** [comment]
Looks good, all fixes seems to be included

**Comment by davidrich27** [comment]
Note: Made a change to the pull request template.  Added a bullet point to PR checklist, to manually run a CI build on PR branch before merging it into main.

File: src/subsplit_dag_storage.hpp
Comment (matsen) [inline src/subsplit_dag_storage.hpp:401]: These should be wrapped in braces then the code reformatted.

**Review by matsen — APPROVED** [review]
@davidrich27 just a few fixes suggested here, but I'm approving so you can merge after those are done.

**Pull request merged**
